docs/sidecar/agent-identities.md

```diff
@@ -0,0 +1,402 @@
+GET /AuthorizationHeader/Graph?AgentIdentity=12345678-1234-1234-1234-123456789012&AgentUserId=87654321-4321-4321-4321-210987654321
+```
+
+**Token Characteristics**:
+- Token type: User token
+- Subject (`sub`): User's object ID
+- Agent identity facet included in token claims
+- Permissions: Delegated permissions scoped to the user
+
+**Use Cases**:
+- Long-running workflows with stored user identifiers
+- Batch operations on behalf of multiple users
+- Systems using object IDs for user reference
+
+### Pattern 4: Regular Request (No Agent)
+
+When no agent parameters are provided, the sidecar uses the incoming token's identity.
+
+**Scenario**: Standard on-behalf-of (OBO) flow without agent identities.
+
+```bash
+GET /AuthorizationHeader/Graph
+```
+
+**Token Characteristics**:
+- Token type: Depends on incoming token and configuration
+- Uses standard OBO or client credentials flow
+- No agent identity facets
+
+## Precedence Rules
+
+When both `AgentUsername` and `AgentUserId` are somehow present (though validation should prevent this), the implementation gives precedence to username:
+
+```
+AgentUsername > AgentUserId
+```
+
+However, this scenario should be avoided and will result in a validation error in strict validation mode.
+
+## Microsoft Entra ID Configuration
+
+### Prerequisites for Agent Identities
+
+1. **Agent Application Registration**:
+   - Register the parent agent application in Microsoft Entra ID
+   - Configure API permissions for downstream APIs
+   - Set up client credentials (certificate or secret)
+
+2. **Agent Identity Configuration**:
+   - Create agent identity registrations
+   - Configure Federated Identity Credentials (FIC) between the parent agent and agent identities
+   - Assign necessary permissions to agent identities
+
+3. **Application Permissions**:
+   - Grant application permissions for autonomous scenarios
+   - Grant delegated permissions for user delegation scenarios
+   - Ensure admin consent is provided where required
+
+### Example: Configuring an Agent Identity
+
+```bash
+# Create agent identity app registration
+az ad app create --display-name "MyAgent Identity"
+
+# Get the agent identity client ID
+AGENT_IDENTITY_ID=$(az ad app list --display-name "MyAgent Identity" --query [0].appId -o tsv)
+
+# Configure FIC between parent agent and agent identity
+az ad app federated-credential create \
+  --id $PARENT_AGENT_APP_ID \
+  --parameters '{
+    "name": "MyAgentIdentityFIC",
+    "issuer": "https://login.microsoftonline.com/<tenant-id>/v2.0",
+    "subject": "'$AGENT_IDENTITY_ID'",
+    "audiences": ["api://AzureADTokenExchange"]
+  }'
+
+# Grant permissions to the agent identity
+az ad app permission add \
+  --id $AGENT_IDENTITY_ID \
+  --api 00000003-0000-0000-c000-000000000000 \
+  --api-permissions e1fe6dd8-ba31-4d61-89e7-88639da4683d=Scope
+
+# Grant admin consent
+az ad app permission admin-consent --id $AGENT_IDENTITY_ID
+```
+
+## Token Claims
+
+### Autonomous Agent Token Claims
```

Comment:
We need to remove this example and point to the (future) Agentic identity platform documentation